From 6e8b50d319c9900585f66a8a2f79bad4c691c4f3 Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Wed, 2 Oct 2024 09:08:15 +0200
Subject: [PATCH] drm/stm: ltdc: Check the security of layer 3.

Split accesses to main node (common access) and sub-nodes which contains
specific features (layer1/2, layer3, rotation).

Change-Id: Ica1a4e17a8783009bcdbf91db32745902f2ba2f1
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/409112
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 drivers/gpu/drm/stm/ltdc.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/stm/ltdc.c b/drivers/gpu/drm/stm/ltdc.c
index b3a0dca98340..2a25119cd995 100644
--- a/drivers/gpu/drm/stm/ltdc.c
+++ b/drivers/gpu/drm/stm/ltdc.c
@@ -2231,14 +2231,14 @@ static int ltdc_get_caps(struct drm_device *ddev)
 		if (ret)
 			return ret;
 
-		np = of_get_child_by_name(dev->of_node, "l0l1");
+		np = of_get_child_by_name(dev->of_node, "l1l2");
 		if (np) {
 			ret = stm32_firewall_get_firewall(np, &ldev->firewall[1], 1);
 			if (ret)
 				return ret;
 		}
 
-		np = of_get_child_by_name(dev->of_node, "l2");
+		np = of_get_child_by_name(dev->of_node, "l3");
 		if (np) {
 			ret = stm32_firewall_get_firewall(np, &ldev->firewall[2], 1);
 			if (ret)
@@ -2265,7 +2265,7 @@ static int ltdc_get_caps(struct drm_device *ddev)
 					 * (do not create a plan)
 					 * if this one is reserved for secure application.
 					 */
-					if (!strcmp("l2", fwl[i].entry)) {
+					if (!strcmp("l3", fwl[i].entry)) {
 						ldev->caps.nb_layers--;
 					} else {
 						stm32_firewall_release_access(fwl);
-- 
2.39.5

